src/analysis/pluginDeclaration.test.js

```diff
@@ -52,4 +55,29 @@ describe("analysis/pluginDeclaration", () => {
       expect(expected).toEqual(actual);
     });
   });
+
+  describe("to/fromJSON", () => {
+    it("works round-trip on an empty declaration", () => {
+      const json = toJSON([emptyDeclaration]);
+      const result = fromJSON(json);
+      expect(result).toEqual([emptyDeclaration]);
+    });
+    it("snapshots on an empty declaration", () => {
+      // stringify to avoid having literal NUL bytes in our source.
+      expect(stringify(toJSON([emptyDeclaration]))).toMatchInlineSnapshot(
+        `"[{\\"type\\":\\"sourcecred/pluginDeclarations\\",\\"version\\":\\"0.1.0\\"},[{\\"edgePrefix\\":\\"E\\\\u0000\\",\\"edgeTypes\\":[],\\"name\\":\\"empty\\",\\"nodePrefix\\":\\"N\\\\u0000\\",\\"nodeTypes\\":[],\\"userTypes\\":[]}]]"`
+      );
+    });
+    it("works round-trip on an non-empty declaration", () => {
+      const json = toJSON([nonEmptyDeclaration]);
+      const result = fromJSON(json);
+      expect(result).toEqual([nonEmptyDeclaration]);
+    });
+    it("snapshots on an non-empty declaration", () => {
+      // stringify to avoid having literal NUL bytes in our source.
+      expect(stringify(toJSON([nonEmptyDeclaration]))).toMatchInlineSnapshot(
+        `"[{\\"type\\":\\"sourcecred/pluginDeclarations\\",\\"version\\":\\"0.1.0\\"},[{\\"edgePrefix\\":\\"E\\\\u0000\\",\\"edgeTypes\\":[{\\"backwardName\\":\\"is pointed to\\",\\"defaultWeight\\":{\\"backwards\\":3,\\"forwards\\":2},\\"description\\":\\"a type\\",\\"forwardName\\":\\"points\\",\\"prefix\\":\\"E\\\\u0000edge\\\\u0000\\"}],\\"name\\":\\"non-empty\\",\\"nodePrefix\\":\\"N\\\\u0000\\",\\"nodeTypes\\":[{\\"defaultWeight\\":2,\\"description\\":\\"a type\\",\\"name\\":\\"node\\",\\"pluralName\\":\\"nodes\\",\\"prefix\\":\\"N\\\\u0000node\\\\u0000\\"}],\\"userTypes\\":[]}]]"`
```

Comment:
Nit: suggest not using an inline snapshot.

Inline snapshots are something I interpret as, "I should read these to validate they're testing expected behavior". Since we already have the round-trip for these I think the contents don't need to be visible, as long as they don't change.

Reply:
Changed.